modules/test/playwright/tests/api-builder/apiFilter.spec.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {APIResponse, expect as baseExpect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+
+export const test = mergeTests(apiHelpersTest, loginTest);
+
+const expect = baseExpect.extend({
+	toBeSuccessful: (response: APIResponse) => ({
+		message: () =>
+			response.ok()
+				? 'Response is successful'
+				: 'Response is not successful',
+		pass: response.ok(),
+	}),
+	toHaveItems: async (response: APIResponse, expectedItems: object[]) => {
+		baseExpect((await response.json()).items).toEqual(expectedItems);
+
+		return {
+			message: () => 'The items are correct',
+			pass: true,
+		};
+	},
+});
+
+test('can GET with API Filter', async ({apiHelpers}) => {
+	await apiHelpers.featureFlag.updateFeatureFlag('LPS-178642', true);
+
+	const objectDefinition =
+		await apiHelpers.objectAdmin.postRandomObjectDefinition('');
```

Comment:
Perhaps we should modify the `postRandomObjectDefinition` so that the `objectFolderErc` is an optional field with `?`. I think is more elegant than having to pass an empty string.

Reply:
Also, from my experience it is very helpful to understand and more efficient using existing object definitions, such as API Application. But it's just my preference.

Reply:
the objectFolderErc is used in another test and the whole method was created by another person, that is why I didn't touch this part :D

Reply:
> Also, from my experience it is very helpful to understand and more efficient using existing object definitions, such as API Application. But it's just my preference.

can you say more about this? not sure I understand 🤔

Reply:
> the objectFolderErc is used in another test and the whole method was created by another person, that is why I didn't touch this part :D

Don't worry, by making it optional it shouldn't affect the other usages of this method.

Reply:
> can you say more about this? not sure I understand 🤔

So, in the other API Builder tests (except for one) I've used already existing System Object definitions such as API Application for schemas because they're already pretty complex structures but we know their fields external reference code, their relationships...

This is better for performance because you don't need to create and delete Object definitions within the test and will help you when recovering some data because of all the known external reference codes.

But as I said, this is more like an opinion. Go for your approach if you find it easier. :smile:

Reply:
@albertojml I decided to leave custom object definition, as the use case is slightly different with system, and also if anything changes for one of the system object in the future our test won't depend on that